### ISSUE TITLE
Is this a kind of assert ?

### ISSUE THREAD

**Issue by mingodad** [gigablast/open-source-search-engine#198]
Looking through the code of `void gotHttpReply2` on `Msg13.cpp` I found this and it seems like a kind of poor man assert:
```
if ( replySize > replyAllocSize ) { char *xx=NULL;*xx=0; } //!!! here it'll segfault on purpose ?
```
Is this intentional ?

**Comment by tcreek** [comment]
Good luck on getting a response

**Comment by mingodad** [comment]
Do you mean this is a dead project ?

**Comment by tcreek** [comment]
No, it is not dead and gets updates often, it's just the owner(s)/contributors do not bother reading the issues section.

**Comment by mingodad** [comment]
Thank you for reply and advise about it !

**Comment by SaptakBhoumik** [comment]
I believe it is some exception handling method using signals. Note:-Just my assumption.

**Comment by mingodad** [comment]
But that's a segfault I'm not sure any exception handler catch it, that's why I said maybe **poor man assertion**.

**Comment by SaptakBhoumik** [comment]
> But that's a segfault I'm not sure any exception handler catch it, that's why I said maybe **poor man assertion**.

You can use signals to catch it but this is just my assumption because there's better way of handling exception

**Comment by SaptakBhoumik** [comment]
> But that's a segfault I'm not sure any exception handler catch it, that's why I said maybe **poor man assertion**.

You can use signals to catch it but this is just my assumption because there's better way of handling exception

**Comment by davidebaldini** [comment]
The expression ```{ char *xx=NULL;*xx=0; }``` is copy-pasted everywhere in the sources, it is indeed a surrogate for ```assert```.
